File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.py
def yak(num1, num2) :
    answer=[]
    min = num1 if num1<num2 else num2
    for i in range(1, min+1):
        if num1%i==0 and num2%i==0 :
            answer.append(i)
    return max(answer)
def solution(numer1, denom1, numer2, denom2):
    d= yak(denom1, denom2)
    denom3 = denom1*denom2//d
    numer3 = (denom3//denom1)*numer1 + denom3//denom2*numer2
    y = yak(denom3, numer3)
    return [numer3//y, denom3//y]

File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = new int[2];
        int denom = denom1*denom2;
        int numer = numer1*denom2 + numer2*denom1;
        int min = (denom>numer) ? numer : denom;
        for(int i=min;i>0;i--){
            if(numer%i==0 && denom%i==0){
                denom /= i;
                numer /= i;
            }      
        }
        answer[0] = numer;
        answer[1] = denom;
        return answer;
    }
}